src/main/java/com/deloitte/elrr/aggregator/rules/ProcessCredential.java

```diff
@@ -142,16 +128,77 @@ private Credential processCredential(final Activity activity,
         if (credential == null) {
 
             credential = createCredential(activity, startDate, endDate);
+            log.info(CREDENTIAL_MESSAGE + " " + credential.getIdentifier()
+                    + " created.");
 
         } else {
 
-            log.info(CREDENTIAL_MESSAGE + " " + activity.getId() + " exists.");
             credential = updateCredential(credential, activity, endDate);
+            log.info(CREDENTIAL_MESSAGE + " " + credential.getIdentifier()
+                    + " updated.");
+
         }
 
         return credential;
     }
 
+    /**
+     * @param context
+     * @param person
+     * @param startDate
+     * @param expires
+     * @return credential
+     * @throws AggregatorException
+     * @throws URISyntaxException
+     */
+    public List<Credential> processAssignedCredentials(final Context context,
```

Comment:
it seems like the only thing you should need here is context, why does this have so many params.